### PR TITLE
Update gems and fix specs

### PR DESCRIPTION
**Pull request by hugomaiavieira**

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://service.update-gems-and-fix-specs.got-board-team-got-board-backend-d07a28673e235c051bbc.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://service.update-gems-and-fix-specs.got-board-team-got-board-backend-d07a28673e235c051bbc.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

File: app/interactions/users/authenticate_user.rb [hunk 1]
```diff
@@ -5,11 +5,6 @@ class AuthenticateUser < ActiveInteraction::Base
   string :token
 
   def execute
-    user = User.find_or_initialize_by(email: email)
-    user.first_name = first_name
-    user.last_name = last_name
-    user.token = token
-    user.save
-    user
+    User.find_by(email: email) || User.create(inputs)
```
Comment (tauil) [inline app/interactions/users/authenticate_user.rb]: If the user exists, it's not going to update the token. The expected behaviour is to update the token always.

File: spec/controllers/api/v1/users_controller_spec.rb [hunk 2]
```diff
@@ -2,10 +2,10 @@
 
 RSpec.describe Api::V1::UsersController, type: :controller do
 
-  describe "GET #autheticate" do
-    it "returns http success" do
-      get :autheticate
-      expect(response).to have_http_status(:success)
+  describe "POST #authenticate" do
+    xit "returns http success" do
```
Comment (tauil) [inline spec/controllers/api/v1/users_controller_spec.rb:6]: `xit` ?

File: spec/factories/players.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 FactoryGirl.define do
   factory :player do
-    match ""
-house "MyString"
+    association :match
```
Comment (tauil) [inline spec/factories/players.rb:3]: If match doesn't exists it creates automatically? Didn't know this option.

Comment (tauil) [inline spec/factories/players.rb:3]: Actually it creates a new match always, right?

File: spec/interactions/users/authenticate_user_spec.rb [hunk 4]
```diff
@@ -2,21 +2,26 @@
...
-  context 'when user exists' do
-    it 'returns user'
+    it 'finds for a user' do
+      is_expected.to eq(user)
+    end
   end
 
   context 'when user does not exists' do
-    it 'creates a new user'
+    let(:attrs) { { 'first_name' => 'John', 'last_name' => 'Doe', 'email' => 'johndoe@mail.com', 'token' => 'some' } }
```
Comment (tauil) [inline spec/interactions/users/authenticate_user_spec.rb]: Just to be consistent, the token should match 'Bearer hex_string'. It's not in the code, but as far as I remember, Ember Simple Auth or Tori sends this 'Bearer ' at the begining of the header authentication string.

File: spec/controllers/api/v1/users_controller_spec.rb (hunk 2 above)
Comment (hugomaiavieira) [inline spec/controllers/api/v1/users_controller_spec.rb:6]: yes, my fix for this was make it pending :shaved_ice:

File: spec/factories/players.rb (hunk 3 above)
Comment (hugomaiavieira) [inline spec/factories/players.rb:3]: Yes, if you do not pass a previously created one, it creates, no matter if using `create` or `build`.

File: spec/interactions/users/authenticate_user_spec.rb (hunk 4 above)
Comment (hugomaiavieira) [inline spec/interactions/users/authenticate_user_spec.rb]: @tauil ok, but this token comes from Google. The preceding 'Bearer' is given when authenticating, so it should appear in the controller specs (the `xit` one), right?

File: spec/factories/players.rb (hunk 3 above)
Comment (tauil) [inline spec/factories/players.rb:3]: Great! :+1:

File: spec/interactions/users/authenticate_user_spec.rb (hunk 4 above)
Comment (tauil) [inline spec/interactions/users/authenticate_user_spec.rb]: I thought that putting it in this attributes would be enough.

File: spec/controllers/api/v1/users_controller_spec.rb (hunk 2 above)
Comment (tauil) [inline spec/controllers/api/v1/users_controller_spec.rb:6]: why don;t remove the block instead? Like this:

```
describe 'POST #authenticate' do
  it 'returns http success'
end
```

File: app/interactions/users/authenticate_user.rb (hunk 1 above)
Comment (tauil) [inline app/interactions/users/authenticate_user.rb]: This is the more important comment of the PR and was left without answer. :crying_cat_face:

Comment (hugomaiavieira) [inline app/interactions/users/authenticate_user.rb]: That was awswered [with code](https://github.com/got-board-team/got-board-api/pull/11/commits/2337b2b29fc954cb26e325819ecf9274549d957e) :wink:

Comment (tauil) [inline app/interactions/users/authenticate_user.rb]: Great. I don't know why, but I didn't got any notification of this code that you linked, but anyway :+1:

File: spec/controllers/api/v1/users_controller_spec.rb (hunk 2 above)
Comment (tauil) [inline spec/controllers/api/v1/users_controller_spec.rb:6]: What about this?

**Pull request merged**